tests/e2e/suites.go

```diff
@@ -0,0 +1,288 @@
+	e2econfig "github.com/vesoft-inc/nebula-operator/tests/e2e/config"
+)
+
+func setupSuite() {
+	framework.Logf("running BeforeSuite actions on node 1")
+
+	ginkgo.By("Add NebulaCluster scheme")
+	err := v1alpha1.AddToScheme(scheme.Scheme)
+	framework.ExpectNoError(err, "failed to add NebulaCluster scheme")
+
+	ginkgo.By("Setup kubernetes")
+	setupKubernetes()
+
+	ginkgo.By("Setup cert manager")
+	setupCertManager()
+
+	ginkgo.By("Setup kruise")
+	setupKruise()
+
+	ginkgo.By("Setup nebula operator")
+	setupNebulaOperator()
+}
+
+func setupSuitePerGinkgoNode() {
+	framework.Logf("running BeforeSuite actions on all nodes")
+}
+
+func cleanupSuite() {
+	framework.Logf("running AfterSuite actions on all nodes")
+	framework.RunCleanupActions()
+
+	ginkgo.By("Cleanup kubernetes")
+	cleanupKubernetes()
+}
+
+func cleanupSuitePerGinkgoNode() {
+	framework.Logf("running AfterSuite actions on node 1")
+}
+
+func setupKubernetes() {
+	if !e2econfig.TestConfig.InstallKubernetes {
+		return
+	}
+	framework.Logf("install kubernetes")
+
+	kindName := e2econfig.TestConfig.KindName
+	if kindName == "" {
+		kindName = e2econfig.DefaultKindName
+	}
+	stdout, _, err := framework.RunCmd("kind", "get", "clusters")
+	framework.ExpectNoError(err, "failed to get kind clusters")
+	isInstall := false
+	sc := bufio.NewScanner(strings.NewReader(stdout))
+	for sc.Scan() {
+		if sc.Text() == kindName {
+			isInstall = true
+		}
+	}
+	if err := sc.Err(); err != nil {
+		framework.ExpectNoError(err)
+	}
+
+	if !isInstall {
+		cmd := []string{"kind", "create", "cluster", "--name", kindName}
+		if framework.TestContext.KubeConfig != "" {
+			cmd = append(cmd, "--kubeconfig", framework.TestContext.KubeConfig)
+		}
+		if e2econfig.TestConfig.KindConfig != "" {
+			cmd = append(cmd, "--config", e2econfig.TestConfig.KindConfig)
+		}
+		_, _, err := framework.RunCmd(cmd[0], cmd[1:]...)
+		framework.ExpectNoError(err, "failed to install kubernetes %s", kindName)
+	}
+
+	clientConfig, err := framework.LoadConfig()
+	framework.ExpectNoError(err, "failed load config")
+	runtimeClient, err := client.New(clientConfig, client.Options{})
+	framework.ExpectNoError(err)
+
+	err = waitForKubernetesWorkloadsReady("", 30*time.Minute, 10*time.Second, runtimeClient)
+	framework.ExpectNoError(err, "failed to wait for kubernetes workloads ready")
+}
+
+func setupCertManager() {
+	if !e2econfig.TestConfig.InstallCertManager {
+		return
+	}
+
+	helmName := "cert-manager" // nolint: goconst
+	helmNamespace := "cert-manager"
+	workloadNamespaces := []string{helmNamespace}
+	helmArgs := []string{
+		"install", helmName,
+		"cert-manager",
+		"--repo", "https://charts.jetstack.io",
+		"--namespace", helmNamespace,
+		"--create-namespace",
+		"--version", "v1.3.1",
+		"--set", "installCRDs=true",
+	}
+	helmInstall(helmName, helmNamespace, workloadNamespaces, helmArgs...)
+}
+
+func setupKruise() {
+	if !e2econfig.TestConfig.InstallKruise {
+		return
+	}
+	helmName := "kruise"
+	helmNamespace := "default"
```

Comment:
set namespace to kruise-system

Reply:
The `helmNamespace` means the helm's namespace nor the crd's namespace to install.

In the helm repo,  there have a namespace resource `kruise-system`. So, if set `helmNamespace` to `kruise-system`, we need to create `kruise-system` first or add `--create-namespce` to helm command.
But it already a namespace `kruise-system` in helm repo.
There's a conflict. I haven't found a better way at the moment.